### ISSUE TITLE
Does not support URL-encoded content in URIs

### ISSUE THREAD

**Issue by jGleitz** [richardszalay/mockhttp#116]
I’m calling an API that uses a URL-encoded path variable (the [AWS IoT HTTPS API](https://docs.aws.amazon.com/iot/latest/developerguide/http.html)). So my request looks like this:

` POST /topics/$aws%2Fthings%2Ftest-host%2Fshadow%2Fname%2Ftest-shadow%2Fupdate`

However, I can only match this request using 
```c#
_mockHttp.Expect(Post, "/topics/$aws/things/test-host/shadow/name/patch_management/update")
```

 This is the wrong expectation, as I also want to test that my code correctly encodes the path variable.

The underlying issue is that [`UrlEncoder` uses `URI.LocalPath`](https://github.com/richardszalay/mockhttp/blob/4536be6b1582157ec33495e783362f0c089e4715/RichardSzalay.MockHttp.Shared/Matchers/UrlMatcher.cs#L78), which decodes and URL-encoding (as all URI accessors do).

**Comment by richardszalay** [comment]
Hi @jGleitz, let me take a look at this. I remember that Android had some issues that required the use of LocalPath so I'll need to be careful with a fix here.

**Comment by richardszalay** [comment]
It turns out I was wrong about the Android issue as it had nothing to do with `LocalPath`. I'm going to include a fix for this in the next version (7.0.0.0), but in the meantime you can use this as a workaround:

```csharp
_mockHttp.Expect("*")
  .With(req => req.RequestUri.AbsolutePath == "/topics/$aws%2Fthings%2Ftest-host%2Fshadow%2Fname%2Ftest-shadow%2Fupdate")
  ...
```
